### PR TITLE
RESOURCE-236 AWS::ApiGateway::UsagePlan

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]
LGTM

File: docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plans.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+: **Field**: `throttle`
+
+`quotas`
+: The target maximum number of permitted requests per a given unit time interval.
+
+: **Field**: `quota`
+
+`product_codes`
+: The AWS Markeplace product identifier to associate with the usage plan as a SaaS product on AWS Marketplace.
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plans.md:69]: @soumyo13 - These FIVE fields arent required here?:

- throttle.burst_limit
- throttle.rate_limit
- quota.limit
- quota.offset
- quote.period

**Review by dkumaras — COMMENTED** [review]
@soumyo13 - Readme file needs to be reviewed?

Left some queries - pls check.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plan.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+: **Field**: `api_stages[0].stage`
+
+`api_stages_throttles`
+: Map containing method level throttling information for API stage in a usage plan.
+
+: **Field**: `api_stages[0].throttle`
+
+`throttle`
+: A map containing method level throttling information for API stage in a usage plan.
```
[the remark this answers is not in the document]
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plan.md]: This is required in the singular resource.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plans.md (hunk 1 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plans.md:69]: ok

File: docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plan.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_usage_plan.md]: Pls check now

**Pull request merged**